pandas/_libs/tslibs/timezones.pyx

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# cython: profile=False
+
+from pandas.compat import string_types, is_platform_windows
+
+
+cdef extern from "Python.h":
+    Py_ssize_t PY_SSIZE_T_MAX
+
+cdef extern from "datetime.h":
+    void PyDateTime_IMPORT()
+
+# import datetime C API
+PyDateTime_IMPORT
+
+
+import numpy as np
+cimport numpy as np
+from numpy cimport ndarray, int64_t, float64_t
+np.import_array()
+
+
+# dateutil compat
+from dateutil.tz import (tzoffset,
+                         tzlocal as _dateutil_tzlocal,
+                         tzfile as _dateutil_tzfile,
+                         tzutc as _dateutil_tzutc,
+                         tzstr as _dateutil_tzstr)
+
+if is_platform_windows():
+    from dateutil.zoneinfo import gettz as _dateutil_gettz
+else:
+    from dateutil.tz import gettz as _dateutil_gettz
+
+
+from pytz.tzinfo import BaseTzInfo as _pytz_BaseTzInfo
+import pytz
+UTC = pytz.utc
+
+
+from cpython cimport PyBool_Check
+cdef extern from "numpy/ndarrayobject.h":
+    bint PyArray_IsIntegerScalar(object)
+
+cdef inline bint is_integer_object(object obj):
+    # Ported from util (which gets it from numpy_helper) to avoid
+    # direct dependency
+    return (not PyBool_Check(obj)) and PyArray_IsIntegerScalar(obj)
+
+
+cdef int64_t NPY_NAT = np.datetime64('nat').astype(np.int64)
+
+
+#----------------------------------------------------------------------
+# time zone conversion helpers
+
+def get_timezone(tz):
+    return _get_zone(tz)
+
+
+cdef bint _is_utc(object tz):
```

Comment:
I think these can all be inline (though if they give warning then prob not)

Reply:
will change.